client/components/follow-button/button.jsx

```diff
@@ -1,17 +1,17 @@
 /**
 * External dependencies
 */
-import React from 'react';
+import React, { PropTypes } from 'react';
 import { noop } from 'lodash';
 
 const FollowButton = React.createClass( {
 
 	propTypes: {
-		following: React.PropTypes.bool.isRequired,
-		onFollowToggle: React.PropTypes.func,
-		iconSize: React.PropTypes.number,
-		tagName: React.PropTypes.string,
-		disabled: React.PropTypes.bool
+		following: PropTypes.bool.isRequired,
+		onFollowToggle: PropTypes.func,
+		iconSize: PropTypes.number,
+		tagName: PropTypes.oneOfType( [ PropTypes.string, PropTypes.func ] ),
```

Comment:
I would suggest a future janitorial that splits this out into `tagName` and `getTagName`